File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans=0,l=0,r=0;
        while(r<prices.size()){
            if(prices[l]<prices[r]){
                ans=max(ans,prices[r]-prices[l]);
            }
            else{
                l=r;
            }
            r++;
        }
        return ans;
    }
};

File: Data Structures & Algorithms/eating-bananas/submission-0.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l,r=1,max(piles)
        ans=r
        while l<=r:
            mid=(l+r)//2
            sum=0
            for i in piles:
                sum+=math.ceil(i/mid)
            if sum<=h:
                r=mid-1
                ans=mid
            else:
                l=mid+1
        return ans